app/controllers/approve_requests_controller.rb

```diff
@@ -3,8 +3,11 @@ class ApproveRequestsController < ApplicationController
   before_action :correct_request, only: [:update, :show]
 
   def index
-    @approval_requests = current_division.approval_requests.paginate(page: params[:page],
-                                    per_page: Settings.requests.per_page)
+    if params[:date_form] && params[:date_to]
```

Comment:
ko nên check ở controller lặp code em nha. với chỗ này đang hạn chế là search 1 lúc 2 tham số, có khi người ta tìm theo 1 tiêu chí thì ko được. Nên define một method instance ở model approval_request để search, ví dụ:
```suggestion
     def filter_between_date date_from, date_to
      if date_from.present? && date_to.present?
        where "created_at between ? and ?", date_from, date_to
      elsif date_from.present? && date_to.blank?
        where "date_from >= ?", date_from
      elsif date_from.blank? && date_to.present?
        where "end_date <= ?", date_to
      elsif date_form.blank? && date_to.blank?
        all
      end
    end
```
```suggestion
  @requests = current_division.approval_requests.filter_between_date(params[:date_form], params[:date_to]).paginate(page: params[:page], per_page: Settings.requests.per_page)
```